### PR TITLE
[C, iOS, AND, UWP] Color for Picker.Title

### LINKED ISSUE #4669: Color for picker title
## Summary
Add the property TitleColor for Pickers to be like the PlaceholderColor in Entries.

## API Changes
New attribute for Picker called TitleColor (see Entry.PlaceholderColor)

XAML:
```
<Picker TitleColor="#EAEAEA" />
// or
<Picker TitleColor="Gray" />
```

C#:

```
var picker = new Picker();
picker.TitleColor = Color.Gray;
// or
picker.TitleColor = Color.FromHex("#EAEAEA");
```

## Intended Use Case
Use the property Picker.TitleColor to harmony with the others interface element colors eliminating the necessity to create a Custom Renderer.

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change ###

The user is now able to set a color for the Picker.Title property.

### Issues Resolved ### 

- fixes #4669

### API Changes ###

Added:
 - Color Picker.TitleColor { get; set; } //Bindable Property

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android
- UWP

As far as I can tell, other platforms do not show a title/placeholder at this time

### Behavioral/Visual Changes ###

When using this new property, the title, as visible on the respective platform, will adhere to the color set by the user.

### Before/After Screenshots ### 

Android

Android popup

UWP

iOS

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs [hunk 1]
```diff
@@ -23,6 +23,10 @@ protected override void Build(StackLayout stackLayout)
 			var titleContainer = new ViewContainer<Picker>(Test.Picker.Title, new Picker());
 			titleContainer.View.Title = "Title";
 
+			var titleColorContainer = new ViewContainer<Picker>(Test.Picker.Title, new Picker());
```
Comment (paymicro) [inline Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs]: The title is the same as the previous one. Could you add `Color` to title?

File: Xamarin.Forms.Platform.UAP/PickerRenderer.cs [hunk 2]
```diff
@@ -210,7 +210,15 @@ void UpdateTextColor()
 
 		void UpdateTitle()
 		{
-			Control.Header = Element.Title;
+			if (Element.TitleColor == default(Color))
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply Xamarin.Forms.Platform.UAP/PickerRenderer.cs]: Cool! Didn't know IsSet, but then the if-statement should be inverted, right?

Either `Element.TitleColor == default(Color)` or `!Element.IsSet(Picker.TitleColorProperty)`

File: Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs (hunk 1 above)
Comment (jfversluis) [reply Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs]: Done

File: Xamarin.Forms.Platform.UAP/PickerRenderer.cs [hunk 3]
```diff
@@ -210,7 +210,15 @@ void UpdateTextColor()
...
+			{
+				Control.Header = Element.Title;
+			}
+			else
+			{
+				Control.HeaderTemplate = (Windows.UI.Xaml.DataTemplate)Windows.UI.Xaml.Application.Current.Resources["ComboBoxHeader"];
+				Control.DataContext = Element;
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply Xamarin.Forms.Platform.UAP/PickerRenderer.cs]: It seems the `HeaderTemplate` takes precedence over `Header`. So now setting `HeaderTemplate` to null whenever the color is default

File: Xamarin.Forms.Platform.Android/AppCompat/PickerRenderer.cs [hunk 4]
```diff
@@ -141,6 +141,9 @@ void UpdatePicker()
 		{
 			Control.Hint = Element.Title;
 
+			if (Element.TitleColor != default(Color))
```
Comment (paymicro) [inline Xamarin.Forms.Platform.Android/AppCompat/PickerRenderer.cs]: here and in other places replace please on `Element.IsSet(Picker.TitleColorProperty)`

File: Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs [hunk 5]
```diff
@@ -23,6 +23,17 @@ protected override void Build(StackLayout stackLayout)
...
 			titleContainer.View.Title = "Title";
 
+			var titleColorContainer = new ViewContainer<Picker>(Test.Picker.TitleColor, new Picker());
+			titleColorContainer.View.Title = "Title Color";
+			titleColorContainer.View.TitleColor = Color.Red;
+
+			var button = new Button() { Text = "Reset color to default" };
+			button.Clicked += (o, a) => {
+				titleColorContainer.View.ClearValue(Picker.TitleColorProperty);
+			};
```
Comment (paymicro) [inline Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs]: extra braces can be removed.

File: Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs [hunk 6]
```diff
@@ -23,6 +23,17 @@ protected override void Build(StackLayout stackLayout)
...
+
+			titleColorContainer.ContainerLayout.Children.Add(button);
```
Comment (paymicro) [inline Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs]: Could you add another button to set the color (ex. Green). To verify that it is successfully set dynamically.

File: Xamarin.Forms.Platform.Android/AppCompat/PickerRenderer.cs (hunk 4 above)
Comment (jfversluis) [reply Xamarin.Forms.Platform.Android/AppCompat/PickerRenderer.cs]: Done

File: Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs (hunk 5 above)
Comment (jfversluis) [reply Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs]: Done

File: Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs (hunk 6 above)
Comment (jfversluis) [reply Xamarin.Forms.Controls/CoreGalleryPages/PickerCoreGalleryPage.cs]: Done

**Comment by paymicro** [comment]
it would be cool if the Android changed title color also in the pop-up dialogue
screenshot_1

**Comment by jfversluis** [comment]
Oe. Challenge accepted!

**Comment by jfversluis** [comment]
Done! The popup title is now also colored 🙂 

Android popup

**Comment by jfversluis** [comment]
Ah sorry, @rookiejava! I thought I looked into Tizen and didn't see any visible title there. But I must've missed something :(

**Comment by rookiejava** [comment]
It's OK. There are so many hidden magic. :)

**Comment by loLiK-CZ** [comment]
UWP implementation changed Picker behavior to have always Title shown when TitleColor is set, even in case when there is no Title value. 
For example: breaks use case, where global style for Picker with TitleColor setter exists, but not everywhere in UI is Picker.Title value set (not expected to show => breaks layout or hides picker and shows empty Title instead)

**Pull request merged**